### ISSUE TITLE
Fscrawler does not delete documents from index on json support

### ISSUE THREAD

**Issue by FrodeRennemo** [dadoonet/fscrawler#237]
Tested on version 2.2. Setting json_support=true, the crawler does not manage to delete documents from the index.

**Comment by dadoonet** [comment]
Thanks for reporting.
Which OS are you using please?

**Comment by FrodeRennemo** [comment]
Using Windows 10. Been looking at the code, looks like the metadata is not being set for json documents.

**Comment by dadoonet** [comment]
TBH the code which is supposed to remove documents is crappy! Can't wait to replace all that stuff by something smarter (like rsync)

**Comment by babadofar** [comment]
json documents aren't deleted since they don't t have the fields path.encoded and file.filename field. That means when fsCrawler tries to figure out what files are already indexed, it gets an empty list back.

**Comment by dadoonet** [comment]
@babadofar exact. I lost that somewhere when I refactored all the code. That's why i said "crappy code". I know it. I wrote it. :)

**Comment by babadofar** [comment]
The code is not that bad @dadoonet! I think it works fine for crawling a not too large file system. At least with some minor bugs out the window 😺 
Do you have any plans on how to use rsync?

**Comment by dadoonet** [comment]
I started a looong time ago to look at https://github.com/perlundq/yajsync (well I contributed a bit there).
So my goal was to have fscrawler server and agent. Basically send modifications over the wire from the agent to the server. The server part would then simply extract with Tika and send to ES.
Of course agent and server are a single application, which means that you don't explicitly need to start an agent and a server.
The server mode can be also activated so any RSYNC app can basically sync files with FSCrawler.

Well, that's what I have in mind. Did not find time to implement all that though.
